### PR TITLE
feat(ui): add action for creating a physical interface

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add support for creating physical interfaces via the api.
- Add selector for fetching the creating interface status.

## QA

N/A

## Fixes

Fixes: #2181.

### CODE REVIEW

File: ui/src/app/store/machine/slice.ts [hunk 1]
```diff
@@ -28,6 +29,24 @@ export type ScriptInput = {
   [x: string]: { url: string };
 };
 
+const generateParams = <P extends { [x: string]: unknown }>(
```
Comment (Caleb-Ellis) [inline ui/src/app/store/machine/slice.ts:32]: Nice. Is the plan to have a type for each action's params? Might be time to break up the types file :grimacing:

**Review by Caleb-Ellis — APPROVED** [review]
LGTM!

File: ui/src/app/store/machine/actions.test.ts [hunk 2]
```diff
@@ -616,6 +617,72 @@ describe("machine actions", () => {
     });
   });
 
+  it("can handle creating a physical interface", () => {
+    expect(
+      actions.createPhysical({
+        enabled: true,
+        interface_speed: "10",
```
Comment (Caleb-Ellis) [inline ui/src/app/store/machine/actions.test.ts:624]: Shouldn't `interface_speed` and `link_speed` be numbers?

File: ui/src/app/store/machine/slice.ts (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/store/machine/slice.ts:32]: That could be helpful, but this at least should remove the need for all of the optional param spreads e.g.
```typescript
...("partitionIds" in params && {
  add_partitions: params.partitionIds,
}),
```

File: ui/src/app/store/machine/actions.test.ts (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/store/machine/actions.test.ts:624]: Thanks, yes.

File: ui/src/app/store/machine/slice.ts (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/machine/slice.ts:32]: Yep that was getting a bit unwieldy.

**Pull request merged**